src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -311,3 +301,31 @@ def purge(self, service_name: str) -> None:
             stderr=subprocess.PIPE,
             timeout=10
         )
+
+    def _haproxy_hosts(self) -> List[str]:
+        # NB: Ideally, we would limit the list to IPs on hosts running
+        # haproxy/ingress only, but due to the nature of cephadm today
+        # we'd "only know the set of haproxy hosts after they've been
+        # deployed" (quoth @adk7398). As it is today we limit the list
+        # of hosts we know are managed by cephadm. That ought to be
+        # good enough to prevent acceping haproxy protocol messages
+        # from "rouge" systems that are not under our control. At
+        # least until we learn otherwise.
+        cluster_ips: List[str] = []
+        for host in self.mgr.inventory.keys():
+            default_addr = self.mgr.inventory.get_addr(host)
+            cluster_ips.append(default_addr)
+            nets = self.mgr.cache.networks.get(host)
+            if not nets:
+                continue
+            for subnet, iface in nets.items():
+                ip_subnet = ipaddress.ip_network(subnet)
+                if ipaddress.ip_address(default_addr) in ip_subnet:
+                    continue  # already present
+                if ip_subnet.is_loopback or ip_subnet.is_link_local:
+                    continue  # ignore special subnets
+                addrs = sum((addr_list for addr_list in iface.values()), [])
```

Comment:
seeing this running mypy locally
```
cephadm/services/nfs.py: note: In member "_haproxy_hosts" of class "NFSService":
cephadm/services/nfs.py:327: error: Need type annotation for "addrs"
```

Reply:
Puzzling. I had the change locally, but I guess I never pushed it... what's extra strange is I didn't see a diff between my local and remote branches. But it was not reflected in the PR. Regardless, I rebased on main and pushed again.